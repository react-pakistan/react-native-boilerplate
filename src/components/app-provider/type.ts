import { ReactNode } from 'react';

export interface IAppProviderProps {
  children : ReactNode;
}
