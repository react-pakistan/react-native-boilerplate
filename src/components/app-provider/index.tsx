import React, { memo, ReactElement } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { appTheme } from '../../theme';
import { IAppProviderProps } from './type';

export const AppProvider = memo(({
  children,
} : IAppProviderProps) : ReactElement => (
  <ThemeProvider theme={appTheme}>
    {children}
  </ThemeProvider>
));
