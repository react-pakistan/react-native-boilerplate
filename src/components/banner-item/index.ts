export * from './banner-item';
export * from './type';
