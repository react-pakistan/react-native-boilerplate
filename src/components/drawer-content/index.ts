export * from './drawer-content';
export * from './type';
