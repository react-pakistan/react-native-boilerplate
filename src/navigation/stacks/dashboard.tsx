import { IEmptyAction } from '@react-pakistan/util-functions/action-creator-factories';
import { DrawerActions } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { Icon } from '@react-pakistan/react-native-icon-collection/icon';
import React, { ReactElement, ReactNode } from 'react';
import { useDispatch } from 'react-redux';
import { hideAppAction } from '../../redux/actions';
import { TYPES } from '../../redux/constants';
import { Dashboard as DashboardScreen } from '../../screens/dashboard';
import { NAVIGATION_ROUTES, NAVIGATION_TITLES } from '../navigation-routes';
import { HeaderIcon } from '../styled';

const Stack = createStackNavigator();

export const DashboardStack = () : ReactElement => {
  // dispatch
  const dispatch = useDispatch();

  return (
    <Stack.Navigator
      screenOptions={{
        headerShown: false,
      }}
    >
      <Stack.Screen
        component={DashboardScreen}
        name={NAVIGATION_ROUTES.TAB_MAIN}
        options={{
          headerRight: () : ReactNode => (
            <HeaderIcon
              onPress={() : IEmptyAction<TYPES.HIDE_APP> => dispatch(hideAppAction())}
            >
              <Icon
                icon='MobileUiGrey9'
              />
            </HeaderIcon>
          ),
          headerLeft: () : ReactNode => (
            <HeaderIcon
              onPress={() : any => DrawerActions.toggleDrawer()}
            >
              <Icon
                icon='MobileUiGrey15'
              />
            </HeaderIcon>
          ),
          headerTitle: NAVIGATION_TITLES.TAB_MAIN,
        }}
      />
    </Stack.Navigator>
  );
};
