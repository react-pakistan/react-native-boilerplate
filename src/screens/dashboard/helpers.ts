import { IProject } from './type';

export const dashboardScreenText = {
  mainHeading: 'React Pakistan Open Source Projects',
};

export const openSourceMainProjects : Array<IProject> = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1565548545/GitHub/React_Pakistan_Storybook_Logo.jpg',
    description: `React Pakistan is an OpenSource & Free Software (FOSS) platform,
      creating React software solutions for the global community.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-main-project-1',
    label: 'React Pakistan',
  },
];

export const openSourceMobileProjects : Array<IProject> = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1548862588/react-native-boilerplate.jpg',
    description: `A collection of curated and well maintained React Native
      boilerplates with various possibilities and combinations.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-1',
    label: 'React Native Boilerplates',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650645/GitHub/react-native-commons-collection.jpg',
    description: `A generic, Commons library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-2',
    label: 'React Native Commons Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650649/GitHub/react-native-ui-collection.jpg',
    description: `A generic, UI library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-3',
    label: 'React Native UI Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    description: `A generic, Icon library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-4',
    label: 'React Native Icon Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    description: `A generic, Logo library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-5',
    label: 'React Native Logo Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    description: `A generic, Logo library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-6',
    label: 'React Native Emoji Collection',
  },
];

export const openSourceWebProjects : Array<IProject> = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1566355017/GitHub/react-commons-collection.jpg',
    description: `A generic, Commons library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-1',
    label: 'React Commons Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1566355010/GitHub/react-ui-collection.jpg',
    description: `A generic, UI library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-2',
    label: 'React UI Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499955/GitHub/react-icons-collection.jpg',
    description: `A generic, Icon library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-3',
    label: 'React Icon Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499957/GitHub/react-logo-collection.jpg',
    description: `A generic, Logo library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-4',
    label: 'React Logo Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499957/GitHub/react-emoji-collection.jpg',
    description: `A generic, Logo library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-5',
    label: 'React Emoji Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499950/GitHub/react-design-collection.jpg',
    description: `A generic, Design library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-6',
    label: 'React Design Collection',
  },
];
