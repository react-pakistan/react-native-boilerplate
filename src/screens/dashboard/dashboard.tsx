import React, { ReactElement } from 'react';
import { FlatList, ListRenderItem } from 'react-native';
import { useTheme } from 'styled-components/native';
import { BannerItem } from '../../components/banner-item';
import { dashboardScreenText, openSourceMainProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';
import { IProject } from './type';

const renderItem : ListRenderItem<IProject> = (
  {
    item: {
      banner,
      description,
      label,
    },
  },
) : ReactElement => (
  <BannerItem
    banner={banner}
    description={description}
    label={label}
  />
);

export const Dashboard = () : ReactElement => {
  // theme
  const theme = useTheme();

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceMainProjects}
        keyExtractor={({ id } : IProject) : string => id}
        ListHeaderComponent={(
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        )}
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
};
