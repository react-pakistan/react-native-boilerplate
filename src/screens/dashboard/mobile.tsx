import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import React, { ReactElement, memo } from 'react';
import { FlatList, ListRenderItem } from 'react-native';
import { withTheme } from 'styled-components/native';
import { BannerItem } from '../../components/banner-item';
import { dashboardScreenText, openSourceMobileProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';
import { IProject } from './type';

export const DashboardMobile = memo(withTheme(({
  theme,
} : IDashboardMobileProps) : ReactElement => {
  const renderItem : ListRenderItem<IProject> = ({ item: { banner, description, label } }) : ReactElement => (
    <BannerItem
      banner={banner}
      description={description}
      label={label}
    />
  );

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceMobileProjects}
        keyExtractor={({ id } : IProject) : string => id}
        ListHeaderComponent={(
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        )}
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
}));

export interface IDashboardMobileProps {
  /**
   *
   */
  theme : IRNTheme;
}
