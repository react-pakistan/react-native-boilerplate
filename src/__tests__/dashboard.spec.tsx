import React from 'react';
import { create } from 'react-test-renderer';
import { AppProvider } from '../components/app-provider';
import { Dashboard } from '../screens/dashboard';

test('renders Dashboard component correctly', () : void => {
  const wrapper = create(
    <AppProvider>
      <Dashboard />
    </AppProvider>,
  ).toJSON();
  expect(wrapper).toMatchSnapshot();
});
