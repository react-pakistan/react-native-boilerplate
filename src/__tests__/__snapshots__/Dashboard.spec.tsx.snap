// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard component correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <Styled(Styled(View))>
        <Styled(Styled(Text))
          color="#2E2E2E"
          fontFamily="Roboto-Bold"
          fontSize={56}
          letterSpacing={0}
          lineHeight={56}
          textAlign="left"
          textTransform="uppercase"
        >
          React Pakistan Open Source Projects
        </Styled(Styled(Text))>
      </Styled(Styled(View))>
    }
    data={
      [
        {
          "banner": "https://res.cloudinary.com/dq6hflqwx/image/upload/v1565548545/GitHub/React_Pakistan_Storybook_Logo.jpg",
          "description": "React Pakistan is an OpenSource & Free Software (FOSS) platform, creating React software solutions for the global community.",
          "id": "react-pakistan-open-source-main-project-1",
          "label": "React Pakistan",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 32,
                "paddingVertical": 32,
              },
            ]
          }
        >
          <Text
            color="#2E2E2E"
            fontFamily="Roboto-Bold"
            fontSize={56}
            letterSpacing={0}
            lineHeight={56}
            style={
              [
                {
                  "color": "#2E2E2E",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 56,
                  "letterSpacing": 0,
                  "lineHeight": 56,
                  "textAlign": "left",
                  "textTransform": "uppercase",
                },
              ]
            }
            textAlign="left"
            textTransform="uppercase"
          >
            React Pakistan Open Source Projects
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "width": 750,
              },
            ]
          }
        >
          <Image
            height={384}
            source={
              {
                "uri": "https://res.cloudinary.com/dq6hflqwx/image/upload/v1565548545/GitHub/React_Pakistan_Storybook_Logo.jpg",
              }
            }
            style={
              [
                {
                  "height": 384,
                  "width": 750,
                },
              ]
            }
            width={750}
          />
          <View
            style={
              [
                {
                  "paddingBottom": 32,
                  "paddingLeft": 32,
                  "paddingRight": 32,
                  "paddingTop": 32,
                },
              ]
            }
          >
            <Text
              color="#2E2E2E"
              fontFamily="Roboto-Medium"
              fontSize={44}
              letterSpacing={0}
              lineHeight={56}
              style={
                [
                  {
                    "color": "#2E2E2E",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 44,
                    "letterSpacing": 0,
                    "lineHeight": 56,
                    "textAlign": "left",
                    "textTransform": "none",
                  },
                ]
              }
              textAlign="left"
              textTransform="none"
            >
              React Pakistan
            </Text>
            <Text
              color="#2E2E2E"
              fontFamily="Roboto-Regular"
              fontSize={52}
              letterSpacing={0}
              lineHeight={56}
              style={
                [
                  {
                    "color": "#2E2E2E",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 52,
                    "letterSpacing": 0,
                    "lineHeight": 56,
                    "textAlign": "left",
                    "textTransform": "none",
                  },
                ]
              }
              textAlign="left"
              textTransform="none"
            >
              React Pakistan is an OpenSource & Free Software (FOSS) platform, creating React software solutions for the global community.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
